src/main/java/com/palantir/stash/stashbot/config/ConfigurationPersistenceImpl.java

```diff
@@ -548,19 +548,22 @@ public PullRequestMetadata getPullRequestMetadata(int repoId, Long prId, String
         Long id = pr.getId();
         String fromSha = pr.getFromRef().getLatestCommit().toString();
         String toSha = pr.getToRef().getLatestCommit().toString();
+        Repository repo = pr.getToRef().getRepository();
 
         PullRequestMetadata[] prms = ao.find(PullRequestMetadata.class,
-            "PULL_REQUEST_ID = ? and FROM_SHA = ?", id, fromSha);
+            "REPO_ID = ? and PULL_REQUEST_ID = ? and FROM_SHA = ?", repo.getId(), id, fromSha);
```

Comment:
All credit goes to ActiveObjects for surfacing it; I just bashed my head against the exception until it went away :)